table/block_based/block_based_table_reader.h

```diff
@@ -446,12 +448,25 @@ class BlockBasedTable : public TableReader {
       bool use_cache, bool prefetch, bool pin,
       BlockCacheLookupContext* lookup_context);
 
-  static void SetupCacheKeyPrefix(Rep* rep);
+  static void SetupCacheKeyPrefix(Rep* rep, const std::string& db_session_id,
+                                  uint64_t cur_file_num);
 
   // Generate a cache key prefix from the file
   template <typename TCache, typename TFile>
   static void GenerateCachePrefix(TCache* cc, TFile* file, char* buffer,
-                                  size_t* size) {
+                                  size_t* size,
+                                  const std::string& db_session_id,
+                                  uint64_t cur_file_num) {
+    if (cc != nullptr && cc->IsSecondaryCacheEnabled() &&
+        db_session_id.size() != 0) {
+      // db_session_id is 20 bytes as defined.
+      assert(db_session_id.size() == 20);
+      memcpy(buffer, db_session_id.c_str(), 20);
+      char* end = EncodeVarint64(buffer + 20, cur_file_num);
+      // kMaxVarint64Length is 10 therefore, the prefix is at most 30 bytes.
+      *size = static_cast<size_t>(end - buffer);
+      return;
+    }
```

Comment:
Better to use 'else' here rather than early 'return'. I only recommend early return for handling trivial or exceptional cases.